File: Python For Everyone, Horstmann/Chapter4/P4.18.py
# Write a program that prints a multiplication table, like this:
#     1    2   3   4   5   6   7   8   9   10
#     2    4   6   8   10  12  14  16  18  20
#     3    6   9   12  15  18  21  24  27  30
#     . . .
#     10   20  30  40  50  60  70  80  90  100

for i in range(1, 11):
    for j in range(1, 11):
        prod = i * j
        print("%4d" % prod, end = "")

    print()

File: Java-for-Everyone-Late-Objects-Horstmann/Chapter-4-Loops/P4_18.java
/*Prime numbers. Write a program that prompts the user for an integer and then prints
 out all prime numbers up to that integer. For example, when the user enters 20, the
 program should print
     2
     3
     5
     7
     11
     13
     17
     19
 Recall that a number is a prime number if it is not divisible by any number except 1
 and itself.*/

import java.util.Scanner;

public class P4_18 {
    public static void main(String[] args) {
	Scanner input = new Scanner(System.in);
	System.out.print("Number: ");
	int number = input.nextInt();
	input.close();
	for (int current_number = 2; current_number <= number; current_number++) {
	    boolean prime = true;
	    int test_number = 2;
	    while (prime && test_number < current_number) {
		if (current_number % test_number == 0) {
		    prime = false;		    
		}
		test_number += 1;
	    }
	    if (prime) {
		System.out.println(current_number);
	    }
	}
    }
}
